### PR TITLE
Add executable name as ctest label

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This allows running all input file tests for a particular executable.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: cmake/AddInputFileTests.cmake [hunk 1]
```diff
@@ -26,6 +26,9 @@ function(add_single_input_file_test INPUT_FILE EXECUTABLE COMMAND_LINE_ARGS
   # Set tags for the test
   set(TAGS "InputFiles;${EXECUTABLE_DIR_NAME};${CHECK_TYPE}")
   string(TOLOWER "${TAGS}" TAGS)
+  # Add the executable name as label, without converting to lower case. This
```
Comment (nilsdeppe) [inline cmake/AddInputFileTests.cmake:29]: This will certainly be useful once we only have one executable per dim :)

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by nilsdeppe** [comment]
This seems to have caused an explosion of labels. Why not just do `-R Poisson -L inputfiles` instead? That's already built into CMake

**Comment by nilsvu** [comment]
Ah yes, I can see how this lead to an explosion of labels (one for each executable, and we have a lot of executables at the moment). My intention here was connected to installations with Spack: You install a particular set of executables, and then only run the tests for those:

```
$ spack install --test=root spectre executables=CharacteristicExtract
# This installs only `CharacteristicExtract` and then runs
# `ctest -L CharacteristicExtract` after the installation
```

**Comment by nilsdeppe** [comment]
Hmm, okay. I'm fine waiting until after we have initial data as a input file parameter to reevaluate. I'm not sure how much that'll help since we've been avoiding extra tests in order to avoid increasing the number of executables built. What is your opinion on changing the thing you pasted to `ctest -R Characteristic -L inputfiles`? Would that be an acceptable alternative to keep test labels somewhat manageable in number?

**Comment by nilsvu** [comment]
Yea we could make the executable name part of the test case name instead of a label, if you prefer. So it would be `ctest -R CharacteristicExtract -L inputfiles`. Maybe labels would still be better because we can associate tests to executables in a more flexible way, and run only the subset of tests for the compiled executables, but I have no strong opinion on this.

**Comment by nilsdeppe** [comment]
Yea, I think having names be `ExecName` and then `-L inputfiles` is a good way of handling it. I would find it much more intuitive to use the exec name. Grepping the repo I can't find the line you pointed to (`git grep "ctest -L"`) Do you know where that needs to be updated?

**Comment by nilsvu** [comment]
https://github.com/spack/spack/blob/8f4e029e3a598a6274e2ee78a40dc97bd6a2a107/var/spack/repos/builtin/packages/spectre/package.py#L261

**Pull request merged**